src/main/java/org/kuali/kfs/gl/batch/service/impl/PosterServiceImpl.java

```diff
@@ -0,0 +1,1306 @@
+            if (StringUtils.isBlank(a21SubAccount.getIndirectCostRecoveryTypeCode()) || ObjectUtils.isNull(a21SubAccount.getIndirectCostRecoveryType())) {
+                String errorFieldName = dataDictionaryService.getAttributeShortLabel(A21SubAccount.class, KFSPropertyConstants.INDIRECT_COST_RECOVERY_TYPE_CODE);
+                String warningMessage = MessageFormat.format(warningMessagePattern, errorFieldName, subAccountBOLabel, subAccountValue, accountBOLabel, accountValue);
+                reportWriterService.writeError(et, new Message(warningMessage, Message.TYPE_WARNING));
+                subAccountOK = false;
+            }
+
+            if (StringUtils.isBlank(a21SubAccount.getFinancialIcrSeriesIdentifier())) {
+                Map<String, Object> icrRatePkMap = new HashMap<String, Object>();
+                icrRatePkMap.put(KFSPropertyConstants.UNIVERSITY_FISCAL_YEAR, et.getUniversityFiscalYear());
+                icrRatePkMap.put(KFSPropertyConstants.FINANCIAL_ICR_SERIES_IDENTIFIER, a21SubAccount.getFinancialIcrSeriesIdentifier());
+                IndirectCostRecoveryRate indirectCostRecoveryRate = businessObjectService.findByPrimaryKey(IndirectCostRecoveryRate.class, icrRatePkMap);
+                if (indirectCostRecoveryRate == null) {
+                    String errorFieldName = dataDictionaryService.getAttributeShortLabel(A21SubAccount.class, KFSPropertyConstants.FINANCIAL_ICR_SERIES_IDENTIFIER);
+                    String warningMessage = MessageFormat.format(warningMessagePattern, errorFieldName, subAccountBOLabel, subAccountValue, accountBOLabel, accountValue);
+                    reportWriterService.writeError(et, new Message(warningMessage, Message.TYPE_WARNING));
+                    subAccountOK = false;
+                }
+            }
+
+            if (activeICRAccounts.isEmpty() || ObjectUtils.isNull(refreshSubAccount)) {
+                String errorFieldName = dataDictionaryService.getAttributeShortLabel(A21IndirectCostRecoveryAccount.class, KFSPropertyConstants.ICR_CHART_OF_ACCOUNTS_CODE) + "/" + dataDictionaryService.getAttributeShortLabel(A21IndirectCostRecoveryAccount.class, KFSPropertyConstants.ICR_ACCOUNT_NUMBER);
+                String warningMessage = MessageFormat.format(warningMessagePattern, errorFieldName, subAccountBOLabel, subAccountValue, accountBOLabel, accountValue);
+                reportWriterService.writeError(et, new Message(warningMessage, Message.TYPE_WARNING));
+                subAccountOK = false;
+            }
+
+            if (subAccountOK) {
+                IndirectCostRecoveryGenerationMetadata metadata = new IndirectCostRecoveryGenerationMetadata(a21SubAccount.getIndirectCostRecoveryTypeCode(),
+                        a21SubAccount.getFinancialIcrSeriesIdentifier());
+
+                List<CuIndirectCostRecoveryAccountDistributionMetadata> icrAccountList = metadata.getAccountLists();
```

Comment:
The PosterServiceImpl has already been customized for Cornell in edu.cornell.kfs.gl.batch.service.impl.CuPosterServiceImpl. Is it possible to override the retrieveSubAccountIndirectCostRecoveryMetadata method in CuPosterServiceImpl instead of bringing in the whole class?